waltz-data/src/main/java/com/khartec/waltz/data/survey/SurveyQuestionResponseDao.java

```diff
@@ -122,29 +150,80 @@ public void saveResponse(SurveyInstanceQuestionResponse response) {
                 .where(SURVEY_QUESTION_RESPONSE.SURVEY_INSTANCE_ID.eq(response.surveyInstanceId())
                         .and(SURVEY_QUESTION_RESPONSE.QUESTION_ID.eq(response.questionResponse().questionId()))));
 
-        Boolean responseExists = dsl.select(when(responseExistsCondition, true).otherwise(false))
+        // save survey_question_response record
+        Boolean responseExists = dsl
+                .select(when(responseExistsCondition, true)
+                        .otherwise(false))
                 .fetchOne(Record1::value1);
 
-        if (responseExists) {
-            dsl.executeUpdate(record);
-        } else {
-            dsl.executeInsert(record);
-        }
+        dsl.transaction(configuration -> {
+            DSLContext txDsl = DSL.using(configuration);
+
+            if (responseExists) {
+                txDsl.executeUpdate(record);
+            } else {
+                txDsl.executeInsert(record);
+            }
+
+            // save survey_question_list_response records
+            response.questionResponse().listResponse().ifPresent(lrs -> {
+                txDsl.deleteFrom(SURVEY_QUESTION_LIST_RESPONSE)
+                        .where(SURVEY_QUESTION_LIST_RESPONSE.SURVEY_INSTANCE_ID.eq(response.surveyInstanceId()))
+                        .and(SURVEY_QUESTION_LIST_RESPONSE.QUESTION_ID.eq(response.questionResponse().questionId()))
+                        .execute();
+
+                if (! lrs.isEmpty()) {
+                    List<SurveyQuestionListResponseRecord> listResponses =
+                            IntStream.range(0, lrs.size())
+                            .mapToObj(i -> {
+                                SurveyQuestionListResponseRecord rec = new SurveyQuestionListResponseRecord();
+                                rec.setSurveyInstanceId(response.surveyInstanceId());
+                                rec.setQuestionId(response.questionResponse().questionId());
+                                rec.setResponse(lrs.get(i));
+                                rec.setPosition(i + 1);
```

Comment:
See other comment about possibly using an `AtomicInt` to remove need for the `IntStream`